File: test/end-to-end/PointClass/NamedPoint.java
//
package pointclass;

import pointclass.point.Point;

public class NamedPoint extends Point {
    private String name;

    public NamedPoint(int x, int y, String name) {
        super(x, y);
        this.name = name;
    }

    public String getLabel() {
        return String.format("%0$s at [%1$d, %2$d] with vector %3$d", this.name, this.x, this.y, this.vector);
    }
}
//

File: test/end-to-end/PointClass/Python/named_point.py
from point import Point

class NamedPoint(Point):

    def __init__(self, x, y, name):
        super().__init__(x, y)
        self.__name = name

    def get_label(self):
        return "{0} at [{1}, {2}] with vector {3}".format(self.__name, self.x, self.y, self._vector)

